synapse/handlers/sso.py

```diff
@@ -790,6 +790,32 @@ def get_mapping_session(self, session_id: str) -> UsernameMappingSession:
         logger.info("Couldn't find session id %s", session_id)
         raise SynapseError(400, "unknown session")
 
+    def revoke_mapping_sessions_for_provider_session_id(
+        self, auth_provider_id: str, auth_provider_session_id: str
+    ) -> None:
+        """Revoke username mapping sessions with the given IdP session ID.
+
+        This is useful to revoke in-flight mapping sessions when a logout notification
+        arrives from the IdP.
+
+        Args:
+            auth_provider_id: A unique identifier for this SSO provider, e.g.
+                "oidc" or "saml".
+            auth_provider_session_id: The session ID got during login from the SSO IdP.
+        """
+        to_delete = []
```

Comment:
Do we actually expect more than 1 session? That seems surprising to me.

Reply:
This is technically possible, although quite unlikely.
Basically, if you start two logins, and do not complete the username mapping sessions, the IDP might return the same `sid` for both cases